.github/workflows/cppcheck-analysis.yml

```diff
@@ -0,0 +1,26 @@
+name: "cppcheck code analysis"
```

Comment:
How about renaming the file to `analyze_target` and use name `Analyze (target)` to allow future additions of other checks that requires elevated permissions?